lib/src/image.rs

```diff
@@ -2,33 +2,112 @@
 //!
 //! APIs for operating on container images in the bootc storage.
 
-use anyhow::{Context, Result};
+use anyhow::{ensure, Context, Result};
 use bootc_utils::CommandRunExt;
+use clap::ValueEnum;
 use fn_error_context::context;
 use ostree_ext::container::{ImageReference, Transport};
+use serde::Serialize;
+use tabled::{settings::Style, Tabled};
 
-use crate::imgstorage::Storage;
+use crate::cli::{ImageListFormat, ImageListType};
 
 /// The name of the image we push to containers-storage if nothing is specified.
 const IMAGE_DEFAULT: &str = "localhost/bootc";
 
+#[derive(Clone, Tabled, Serialize, ValueEnum)]
+enum ImageListTypeColumn {
+    Host,
+    Logical,
+}
+
+impl std::fmt::Display for ImageListTypeColumn {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.to_possible_value().unwrap().get_name().fmt(f)
+    }
+}
+
+#[derive(Tabled, Serialize)]
+struct ImageOutput {
+    #[tabled(rename = "IMAGE TYPE")]
+    image_type: ImageListTypeColumn,
+
+    #[tabled(rename = "IMAGE")]
+    image: String,
+    // TODO: Add hash, size, etc? Difficult because [`ostree_ext::container::store::list_images`]
+    // only gives us the pullspec.
+}
+
+#[context("Listing host images")]
+fn list_host_images(sysroot: &crate::store::Storage) -> Result<Vec<ImageOutput>> {
+    let repo = sysroot.repo();
+    let images = ostree_ext::container::store::list_images(&repo).context("Querying images")?;
+
+    Ok(images
+        .iter()
+        .map(|x| ImageOutput {
+            image: x.to_string(),
+            image_type: ImageListTypeColumn::Host,
+        })
+        .collect())
+}
+
+#[context("Listing logical images")]
+fn list_logical_images(sysroot: &crate::store::Storage) -> Result<Vec<ImageOutput>> {
+    let output = sysroot
+        .get_ensure_imgstore()?
+        .new_image_cmd()?
+        .arg("list")
+        .arg("--format={{.Repository}}:{{.Tag}}")
```

Comment:
Alternatively we could use `--format=json`

Reply:
That was my initial attempt, but that doesn't contain any trivially accessible equivalent of `{{.Repository}}:{{.Tag}}` that I could find. In the Go Template [handling code](https://github.com/containers/podman/blob/72f1617facbb83fbc1e74583f62394dfc6142109/cmd/podman/images/list.go#L190-L193), `Repository` and `Tag` are coming from the [imageReporter](https://github.com/containers/podman/blob/72f1617facbb83fbc1e74583f62394dfc6142109/cmd/podman/images/list.go#L336-L340) struct, but the JSON [handling code dumps a different struct called `image`](https://github.com/containers/podman/blob/72f1617facbb83fbc1e74583f62394dfc6142109/cmd/podman/images/list.go#L164-L168) which, when [populated from `imageReporter`](https://github.com/containers/podman/blob/72f1617facbb83fbc1e74583f62394dfc6142109/cmd/podman/images/list.go#L173-L176), doesn't copy the `Repository` and `Tag` fields